### DISCUSSION TITLE
Listing a bucket does not work, it returns an empty array

### DISCUSSION THREAD

**Discussion by H7KZ** [supabase/supabase#5936] [Questions] [answered]
![supaVýstřižek](https://user-images.githubusercontent.com/74021016/158564506-3ee01af8-f6b4-42cf-a68e-e9901e8224f6.PNG)
So this is my code for listing all the files in storage bucket named files... but It doesn't work... help please?

**Comment by GaryAustin1** [comment, marked as answer]
Do you have policy set for select access for objects?

**Comment by thorwebdev** [comment]
https://supabase.com/docs/guides/storage#security

**Comment by joshdance** [comment]
Is there any links to tutorials on the policies you need set to interact with storage buckets? I have seen this question multiple times. I have read the docs. I can't figure it out. Super frustrating. :)

**Comment by 1991sig** [comment]
In case anyone else finds this thread, I can share the solution I found after being stuck for a couple of hours:

## Step 1 - Enable read access to `storage.buckets`:

Make a new policy under here giving read access (or more if you want):
![image](https://github.com/supabase/supabase/assets/40396857/893e6cf2-7afc-428c-8ecc-aa3d664bdb88)

This will enable reading from the `storage.buckets` table, like this with the client in TypeScript:
```typescript
let { data, error } = await supabase.storage.listBuckets();
```

## Step 2 - Enable read access to a specific bucket:

Make a read policy fitting your needs for the specific storage bucket under here:
![image](https://github.com/supabase/supabase/assets/40396857/79e2626c-a96b-45fb-b749-71218858b2e6)


This will enable reading rows from the `storage.objects` for that specific bucket, like this:
```typescript
let { data, error } = await supabase.storage.from("my_bucket").list();
```
